server/helpers/wowCommunityApi.js

```diff
@@ -38,33 +39,145 @@ const get = (url, metricLabels) => {
     },
     onFailure: err => {
       if (err instanceof RequestTimeoutError) {
-        end({ statusCode: 'timeout' });
+        end({
+          statusCode: 'timeout',
+        });
       } else if (err instanceof RequestSocketTimeoutError) {
-        end({ statusCode: 'socket timeout' });
+        end({
+          statusCode: 'socket timeout',
+        });
       } else if (err instanceof RequestConnectionResetError) {
-        end({ statusCode: 'connection reset' });
+        end({
+          statusCode: 'connection reset',
+        });
       } else if (err instanceof RequestUnknownError) {
-        end({ statusCode: 'unknown' });
+        end({
+          statusCode: 'unknown',
+        });
       } else {
-        end({ statusCode: err.statusCode });
+        end({
+          statusCode: err.statusCode,
+        });
       }
     },
     onSuccess: () => {
-      end({ statusCode: 200 });
+      end({
+        statusCode: 200,
+      });
     },
   });
 };
-const makeBaseUrl = region => `https://${region}.api.battle.net/wow`;
+const makeBaseUrl = region => `https://${region}.api.blizzard.com`;
+
+const getAccessToken = async region => {
+  let end;
+  if (!!clientToken[region] && !!clientToken[region].accessToken && clientToken[region].expires > new Date()) {
+    return clientToken[region].accessToken;
+  }
+
+  const url = `https://${region}.battle.net/oauth/token?grant_type=client_credentials&client_id=${process.env.BATTLE_NET_API_CLIENT_ID}&client_secret=${
+    process.env.BATTLE_NET_API_CLIENT_SECRET
+  }`;
+
+  const tokenRequst = await retryingRequest({
+    url,
+    headers: {
+      'User-Agent': USER_AGENT,
+    },
+    gzip: true,
+    // we'll be making several requests, so pool connections
+    forever: true,
+    timeout: 4000,
+    shouldRetry: error => {
+      return true;
+    },
+    onBeforeAttempt: () => {
+      end = blizzardApiResponseLatencyHistogram.startTimer({
+        category: 'access_token',
+        region,
+      });
+    },
+    onFailure: err => {
+      if (err instanceof RequestTimeoutError) {
+        end({
+          statusCode: 'timeout',
+        });
+      } else if (err instanceof RequestSocketTimeoutError) {
+        end({
+          statusCode: 'socket timeout',
+        });
+      } else if (err instanceof RequestConnectionResetError) {
+        end({
+          statusCode: 'connection reset',
+        });
+      } else if (err instanceof RequestUnknownError) {
+        end({
+          statusCode: 'unknown',
+        });
+      } else {
+        end({
+          statusCode: err.statusCode,
+        });
+      }
+    },
+    onSuccess: () => {
+      end({
+        statusCode: 200,
+      });
+    },
+  });
+
+  const tokenData = JSON.parse(tokenRequst);
+  let expireDate = new Date();
+  expireDate = expireDate.setSeconds(expireDate.getSeconds() + tokenData.expires_in);
```

Comment:
Could be a one line:

```suggestion
  const expireDate = (new Date()).setSeconds(expireDate.getSeconds() + tokenData.expires_in);
```

Reply:
Had to make the expireDate in the second line a date.now() instead but yeah that works 👍